moveit_ros/perception/handeye_calibration_target/src/handeye_aruco_target.cpp

```diff
@@ -0,0 +1,282 @@
+    if (valid == 0)
+    {
+      ROS_WARN_STREAM_NAMED(LOGNAME, "Cannot estimate aruco board pose.");
+      return false;
+    }
+
+    if (std::log10(std::fabs(rotation_vect_[0])) > 10 || std::log10(std::fabs(rotation_vect_[1])) > 10 ||
+        std::log10(std::fabs(rotation_vect_[2])) > 10 || std::log10(std::fabs(translation_vect_[0])) > 10 ||
+        std::log10(std::fabs(translation_vect_[1])) > 10 || std::log10(std::fabs(translation_vect_[2])) > 10)
+    {
+      ROS_WARN_STREAM_NAMED(LOGNAME, "Invalid target pose, please check CameraInfo msg.");
+      return false;
+    }
+
+    cv::Mat image_rgb;
+    cv::cvtColor(image, image_rgb, cv::COLOR_GRAY2RGB);
+    cv::aruco::drawDetectedMarkers(image_rgb, marker_corners);
+    drawAxis(image_rgb, camera_matrix_, distortion_coeffs_, rotation_vect_, translation_vect_, 0.1);
+    image = image_rgb;
+  }
+  catch (const cv::Exception& e)
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "Aruco target detection exception: " << e.what());
+    return false;
+  }
+
+  return true;
+}
+
+geometry_msgs::TransformStamped HandEyeArucoTarget::getTransformStamped(const std::string& frame_id) const
+{
+  geometry_msgs::TransformStamped transform_stamped;
+  transform_stamped.header.stamp = ros::Time::now();
+  transform_stamped.header.frame_id = frame_id;
+  transform_stamped.child_frame_id = "handeye_target";
+
+  tf2::Quaternion quaternion(0, 0, 0, 1);
+  convertToTFQuaternion(rotation_vect_, quaternion);
+  transform_stamped.transform.rotation.x = quaternion.getX();
+  transform_stamped.transform.rotation.y = quaternion.getY();
+  transform_stamped.transform.rotation.z = quaternion.getZ();
+  transform_stamped.transform.rotation.w = quaternion.getW();
+
+  std::vector<double> translation(3, 0);
+  convertToStdVector(translation_vect_, translation);
+  transform_stamped.transform.translation.x = translation[0];
+  transform_stamped.transform.translation.y = translation[1];
+  transform_stamped.transform.translation.z = translation[2];
+
+  return transform_stamped;
+}
+
+bool HandEyeArucoTarget::convertToTFQuaternion(const cv::Vec3d& input_rvect, tf2::Quaternion& quaternion) const
+{
+  CV_Assert(input_rvect.rows == 3 && input_rvect.cols == 1);  // 3x1 rotation vector
+  cv::Mat rotation_matrix;
+  cv::Rodrigues(input_rvect, rotation_matrix);
+  CV_Assert(rotation_matrix.rows == 3 && rotation_matrix.cols == 3);  // 3x3 rotation matrix
+
+  tf2::Matrix3x3 m;
+  m.setValue(rotation_matrix.ptr<double>(0)[0], rotation_matrix.ptr<double>(0)[1], rotation_matrix.ptr<double>(0)[2],
+             rotation_matrix.ptr<double>(1)[0], rotation_matrix.ptr<double>(1)[1], rotation_matrix.ptr<double>(1)[2],
+             rotation_matrix.ptr<double>(2)[0], rotation_matrix.ptr<double>(2)[1], rotation_matrix.ptr<double>(2)[2]);
+  m.getRotation(quaternion);
+  return true;
+}
+
+bool HandEyeArucoTarget::convertToStdVector(const cv::Vec3d& input_tvect, std::vector<double>& translation) const
```

Comment:
See comment above.

Reply:
Also convert the assert to the error report here as suggested.

Reply:
Looks good.